ntp.go

```diff
@@ -386,7 +384,11 @@ func getTime(host string, opt QueryOptions) (*msg, ntpTime, error) {
 	}
 
 	// Keep track of the time the response was received.
-	recvTime := toNtpTime(time.Now())
+	delta := time.Since(sendTime) // uses monotonic clock in Go 1.9+
+	if delta < 0 {                // that's possible in Go 1.8 and older versions
+		return nil, 0, errors.New("client clock ticked backwards")
+	}
```

Comment:
I wonder if it should also check whether `delta > timeout` (or `delta > timeout * 2`) for the case when the system clock is reset to a distant time in the future.

Reply:
It's hard to chose proper "tolerance", I think that delay may be quite high on CPU-bound system (dozens of seconds), OTOH SNTP clock sync at alike system is close to impossible. Your suggestion makes some sense, but I don't know if suggested implementation is a good idea...
Probably, the _right_ thing to do to avoid scheduler delay is to timestamp incoming packets, but it makes things even more complex.

Reply:
Good point.  Then again, if the scheduler is imposing a delay of more than a few seconds between these time measurements, the resulting `ClockOffset` is also going to be significantly skewed.

Reply:
Yeah. I don't know. Do you want me to add it for this PR to be merged? :)

Reply:
I think it's not super important.  You can add it to another PR if you think it might be useful.